dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -213,6 +235,19 @@ class DLL_PUBLIC VideoDecoderBase : public Operator<Backend> {
       if (boundary_type_ == boundary::BoundaryType::ISOLATED) {
         sequence_len = std::min(sequence_len, GetAvailableNumFrames(sample_idx));
       }
+    } else if (end_frame_.HasValue()) {
+      int start = GetStartFrame(sample_idx);
+      int end = GetEndFrame(sample_idx);
+      if (boundary_type_ == boundary::BoundaryType::ISOLATED) {
+        int end_of_video = GetTotalNumFrames(sample_idx);
+        end = std::min(end, end_of_video);
+      }
+      int stride = GetStride(sample_idx);
+      DALI_ENFORCE(end > start,
+                   make_string("end_frame (", end, ") must be greater than start_frame (", start,
+                               "), for sample #", sample_idx));
+      sequence_len =
+          (end - start + stride - 1) / stride;  // Round up to include all frames in [start,end)
```

Comment:
It could be too much for this PR, just a food for thought - maybe the user can provide seq_len and end frame with stride and skip start frame?

Reply:
maybe in a follow up someday? :)

Reply:
Sure, just food for thought.